rust/connlib/tunnel/src/client.rs

```diff
@@ -434,33 +463,32 @@ impl ClientState {
         }
     }
 
-    /// Clears the DNS resource NAT state for a given domain.
+    /// Recreate the DNS resource NAT state for a given domain.
     ///
-    /// Once cleared, this will trigger the client to submit another `AssignedIp`s event to the Gateway.
+    /// This will trigger the client to submit another `AssignedIp`s event to the Gateway.
     /// On the Gateway, such an event causes a new DNS resolution.
     ///
     /// We call this function every time a client issues a DNS query for a certain domain.
     /// Coupling this behaviour together allows a client to refresh the DNS resolution of a DNS resource on the Gateway
     /// through local DNS resolutions.
-    fn clear_dns_resource_nat_for_domain(&mut self, message: &dns_types::Response) {
-        let mut any_deleted = false;
-
-        self.dns_resource_nat_by_gateway
-            .retain(|(_, candidate), state| {
-                let DnsResourceNatState::Confirmed = state else {
-                    return true;
-                };
-
-                if candidate == &message.domain() {
-                    any_deleted = true;
-                    return false;
+    ///
+    /// We model the [`DnsResourceNatState::Recreating`] state differently from just removing the entry to allow packets
+    /// to continue flowing to the Gateway while the DNS resource NAT is being recreated.
+    /// In most cases, the DNS records will not change and as such, performing this will not interrupt the flow of packets.
```

Comment:
Just to confirm, does musl's dns on the gateway always return the same IP if there are multiple A or AAAA records for a domain? If so would that cause anything to break here?